autoreduce_frontend/reduction_viewer/views/common.py

```diff
@@ -94,56 +113,83 @@ def prepare_arguments_for_render(arguments: ReductionArguments, instrument: str)
     return final_standard, final_advanced, variable_help
 
 
+def fetch_api_urls(vars_kwargs):
+    """Convert file URLs in vars_kwargs into API URL strings."""
+    for category, headings in vars_kwargs.items():
+        for heading, heading_value in headings.items():
+            if "file" in heading.lower() and isinstance(heading_value, dict):
+                try:
+                    vars_kwargs[category][heading]["all_files"] = {}
+                    path = heading_value["url"].partition("master")[2]
+                    url = "https://api.github.com/repos/mantidproject/scriptrepository/contents" + path
+                    vars_kwargs[category][heading]["api"] = url
+                    req = requests.get(url)
+                    data = json.loads(req.content)
+
+                    for link in data:
+                        file_name = link["name"]
+                        url, _, default = link["download_url"].rpartition("/")
+                        auth_token = os.environ.get("AUTOREDUCTION_GITHUB_AUTH_TOKEN", "")
+                        req = requests.get(f"{url}/{default}", headers={"Authorization": f"Token {auth_token}"})
+                        value = req.text
+                        vars_kwargs[category][heading]["all_files"][file_name] = {
+                            "url": url,
+                            "default": default,
+                            "value": value,
+                        }
+                except Exception:
+                    pass
```

Comment:
Also this should be logged as `logger.error` (check logger imports in other files)